lib/Parser/Parse.cpp

```diff
@@ -10473,8 +10473,16 @@ void Parser::FinishDeferredFunction(ParseNodePtr pnodeScopeList)
             pnodeFnc->sxFnc.pnodeVars = nullptr;
             m_ppnodeVar = &pnodeFnc->sxFnc.pnodeVars;
 
+            ParseNodePtr pnodeFncSaveNonLambda = nullptr;
+
+            pnodeFncSaveNonLambda = m_currentNodeNonLambdaFunc;
```

Comment:
`GetCurrentFunctionNode()` returns the lambda function node while inside a lambda. This is probably why `GetCurrentNonLambdaFunctionNode()` was created in the first place.